aten/src/ATen/native/cuda/UpSampleTrilinear3d.cu

```diff
@@ -363,6 +364,10 @@ static void upsample_trilinear3d_backward_out_cuda_template(
                 odata,
                 idata_ptr);
         C10_CUDA_KERNEL_LAUNCH_CHECK();
+
+        if (!grad_input_.is_contiguous()) {
+            grad_input_.copy_(grad_input);
+        }
```

Comment:
This is visible from the outside, which is not correct.
Why do we need it at all?

Reply:
This is what we typically do for backward ops I would say:
https://github.com/pytorch/pytorch/blob/41ffea2f99b91237ed2952c8d94c6b0e56548384/aten/src/ATen/native/cuda/UpSampleBilinear2d.cu#L464-L466

> Why do we need it at all?

grads algorithm only working for contiguous tensors and previously Line 324 is wrong when trying to make grad_input to be contiguous:
```
// A contiguous tensor is required for the kernel launch config
grad_input.contiguous();
```

Reply:
Yes, I understand the bug, and the fix LGTM, but I don't think we should change the values of an input tensor in-place. That will surely break double gradients though that variable.

Reply:
nvm, this is completely fine